feat(Alura): new presence

MadsonMendes opened the pull request:
## Description 
<!-- A clear and detailed description of the changes, referencing issues if applicable -->
Created a presence for Alura as a request in student's server. Added some dynamic elements as well. Planning to translate everything to Portuguese later. 
## Acknowledgements
- [X] I read the [Presence Guidelines](https://github.com/PreMiD/Presences/blob/main/.github/CONTRIBUTING.md)
- [X] I linted the code by running `yarn format`
- [X] The PR title follows the repo's [commit conventions](https://github.com/PreMiD/Presences/blob/main/.github/COMMIT_CONVENTION.md)

## Screenshots
<details>
<summary> Proof showing the creation/modification is working as expected </summary>
<!-- 
    Screenshots of the presence settings (if applicable) and at least TWO screenshots of the presence displaying correctly
    Including these screenshots will assist the reviewing processes thus speeding up the process of the pull request being merged
-->  

![image](https://cdn.discordapp.com/attachments/878447042630942741/997252077543772221/unknown.png)  

![image](https://cdn.discordapp.com/attachments/878447042630942741/997251846429229197/unknown.png)  

![image](https://cdn.discordapp.com/attachments/692756963032956931/997191383184064543/unknown.png)  

![image](https://cdn.discordapp.com/attachments/692756963032956931/997181367018332202/unknown.png)  

![image](https://cdn.discordapp.com/attachments/834111811116466180/997200577257812080/unknown.png)  
</details>

MadsonMendes committed:
feat(Alura): new presence

MadsonMendes committed:
feat(Alura): updating mock information

MadsonMendes committed:
forgot to lint lol

Signed-off-by: Madson Mendes <50425778+MadsonMendes@users.noreply.github.com>

ImRodry commented:
The website is in portuguese so the presence should be too. I don’t think you should have multilang

EncryptedDev commented on websites/A/Alura/presence.ts:
```diff
@@ -0,0 +1,278 @@
+				"div.container.course-header-banner-content > div > div.hreview-aggregate > h1 > strong"
+			)
+		) {
+			title = document.querySelector("section.task-menu-header > div > a > h2");
+			presenceData.state = title.textContent;
```
```suggestion
			presenceData.state = document.querySelector("section.task-menu-header > div > a > h2").textContent;
```

EncryptedDev commented on websites/A/Alura/presence.ts:
```diff
@@ -0,0 +1,278 @@
+			title = document.querySelector("section.task-menu-header > div > a > h2");
+			presenceData.state = title.textContent;
+		} else {
+			title = document.querySelector(
+				"div.container.course-header-banner-content > div > div.hreview-aggregate > h1 > strong"
+			);
+			presenceData.state = title.textContent;
```
```suggestion
			presenceData.state = document.querySelector(
				"div.container.course-header-banner-content > div > div.hreview-aggregate > h1 > strong"
			).textContent;
```

EncryptedDev commented on websites/A/Alura/presence.ts:
```diff
@@ -0,0 +1,278 @@
+		presenceData.largeImageKey = "scubadev";
+		presenceData.details = 'Listening to "ScubaDev"'; // alura.scubadev
+		if (
+			document.querySelector(".podcast-header-title") !== null &&
+			document.querySelector(".--margin-left-image") === null
```
Update this on future references of `!== null` and `=== null`
```suggestion
			document.querySelector(".podcast-header-title") &&
			!document.querySelector(".--margin-left-image")
```

EncryptedDev reviewed (changes requested):
Hello! I've noticed the following imperfections with your PR, please address them :+1:

MadsonMendes committed:
Update presence.ts

removed comments + translated to portuguese + all assets are in imgur + removed unnecessary let item

Signed-off-by: Madson Mendes <50425778+MadsonMendes@users.noreply.github.com>

EncryptedDev reviewed (commented):
Hello 👋

- Run `yarn format` as the metadata file is not formatted with tabs

EncryptedDev committed:
chore: append review

EncryptedDev committed:
chore: append review

MadsonMendes committed:
Updated with tabs

Signed-off-by: Madson Mendes <50425778+MadsonMendes@users.noreply.github.com>

MadsonMendes committed:
Update presence.ts

Signed-off-by: Madson Mendes <50425778+MadsonMendes@users.noreply.github.com>

fr0r commented on websites/A/Alura/dist/metadata.json:
```diff
@@ -0,0 +1,29 @@
+	"service": "Alura",
+	"description": {
+		"en": "The largest online technology school in Brazil.",
+		"pt_BR": "A maior escola online de tecnologia do Brasil."
```
👍

ImRodry commented on websites/A/Alura/presence.ts:
```diff
@@ -0,0 +1,250 @@
+		else if (
+			document
+				.querySelector("a.course-header-banner-breadcrumb__category")
+				.getAttribute("aria-label") === "Inovação & Gestão"
```
This should only be computed once so you should make it into a constant

ImRodry commented on websites/A/Alura/presence.ts:
```diff
@@ -0,0 +1,250 @@
+		)
+			presenceData.largeImageKey = "https://i.imgur.com/mLpf7Yz.png";
+		else if (
+			document.querySelector("div.formacao__info-categoria > a").textContent ===
```
Ditto

ImRodry commented on websites/A/Alura/presence.ts:
```diff
@@ -0,0 +1,250 @@
+		presenceData.details = "Vendo meus cursos";
+	else if (document.location.pathname.includes("/user")) {
+		if (document.querySelector(".profile-header-name")) {
+			presenceData.details = strings.user;
```
Why are strings being used here? That makes no sense

ImRodry commented on websites/A/Alura/presence.ts:
```diff
@@ -0,0 +1,250 @@
+			if (document.location.pathname.includes("/planos-de-estudo"))
+				presenceData.details = "Vendo planos de estudo";
+			else if (document.location.pathname.includes("/payments"))
+				presenceData.details = "Vendo pegamentos";
```
Pegamentos é noutro lado…

ImRodry commented on websites/A/Alura/presence.ts:
```diff
@@ -0,0 +1,250 @@
+			else if (document.location.pathname.includes("/payments"))
+				presenceData.details = "Vendo pegamentos";
+			else document.location.pathname.includes("/rank");
+			presenceData.details = "Vendo o ranking";
```
De quê?

ImRodry commented on websites/A/Alura/presence.ts:
```diff
@@ -0,0 +1,250 @@
+			!document.querySelector(".--margin-left-image")
+		) {
+			presenceData.state = `"${
+				document.querySelector(".podcast-header-title").textContent
```
Don’t compute this twice

ImRodry commented on websites/A/Alura/presence.ts:
```diff
@@ -0,0 +1,250 @@
+		}
+	} else if (document.location.pathname.includes("/cases")) {
+		presenceData.largeImageKey = "https://i.imgur.com/rtevoAV.png";
+		presenceData.details = "Assistindo um Case";
```
Case não deve estar em maiúscula

ImRodry commented on websites/A/Alura/presence.ts:
```diff
@@ -0,0 +1,250 @@
+			document.location.pathname.includes("/extra")
+		) {
+			presenceData.state = `"${
+				document.querySelector(".intro__title").textContent
```
Don’t compute this twice

ImRodry commented on websites/A/Alura/presence.ts:
```diff
@@ -0,0 +1,250 @@
+		}
+	} else if (document.location.pathname.includes("/hipsterstech")) {
+		presenceData.largeImageKey = "https://i.imgur.com/T7SMsQU.png";
+		presenceData.details = 'Ouvindo o "Hipsters.tech"';
```
O outro tinha T maiúsculo e este já não?

ImRodry reviewed (commented):
Here’s a proper review of the strings + code. I didn’t comment on all lines but there is a lot of unnecessary computation being done that should be avoided

MadsonMendes committed:
Changed my account ID + Removed multilanguage

Signed-off-by: Madson Mendes <50425778+MadsonMendes@users.noreply.github.com>

MadsonMendes committed:
Update presence.ts

Signed-off-by: Madson Mendes <50425778+MadsonMendes@users.noreply.github.com>

EncryptedDev commented:
Ignore the comments, my apologies
